bokeh/command/subcommands/serve.py

```diff
@@ -311,6 +311,20 @@
     DEFAULT_LOG_FORMAT=DEFAULT_LOG_FORMAT
 )
 
+def _fixup_deprecated_host_args(args):
+    if args.host is not None and len(args.host) > 0:
+        if args.allow_websocket_origin is None:
+            args.allow_websocket_origin = []
+        args.allow_websocket_origin += args.host
+        args.allow_websocket_origin = list(set(args.allow_websocket_origin))
+        warnings.warn(
+            "The --host parameter is deprecated because it is no longer needed. "
+            "It will be removed and trigger an error in a future release. "
+            "Values set now will be copied to --allow-websocket-origin. "
+            "Depending on your use case, you may need to set current --host "
+            "values for 'allow_websocket_origin' instead."
+        )
```

Comment:
I wonder if it'd be nicer to leave this undeprecated but right now its only function would be to update the allowed socket origins. In the future, it could hypothetically do something else, who knows. The difference between `--host` and `--allow-websocket-origin` is that `--host` expresses an intent ("this is the public URL of this app") while `--allow-websocket-origin` expresses more of a micromanaged implementation detail. So if we ever wanted to do something else with the knowledge of our public hostname besides websocket origin, we'd add that to `--host` but not to `--allow-websocket-origin`. Not sure.

Full disclosure https://github.com/Anaconda-Platform/anaconda-project currently provides `--host` to Bokeh apps, and to avoid this warning while still working with old Bokeh I'll have to run `bokeh --version` or something preflight before running the app itself. Which I can do and it isn't a giant deal but it is a little simpler if `--host` keeps working...